### ISSUE TITLE
--pattern

### ISSUE THREAD

**Issue by miniHive** [c-cube/datalog#3]
It seems the command-line argument in

**$ datalog_cli tests/clique10.pl --pattern 'same_clique(1,X)'**

should be

**$ datalog_cli tests/clique10.pl -pattern 'same_clique(1,X)'**

with a single dash.

**Comment by c-cube** [comment]
works for me:


```
$ ls -l datalog_cli.exe
lrwxrwxrwx 1 simon simon 48 Jan  3 08:16 datalog_cli.exe -> _build/default/src/bottom_up_cli/datalog_cli.exe*
$ ./datalog_cli.exe tests/clique10.pl  --pattern 'same_clique(1,X)'
% start datalog
% parse file tests/clique10.pl
% process 15 clauses
% computing fixpoint...
% done.
% facts matching pattern same_clique(1, X0):
  same_clique(1, 4).
  same_clique(1, 5).
  same_clique(1, 3).
  same_clique(1, 2).
  same_clique(1, 1).
  same_clique(1, 0).
% max_heap_size: 491520; minor_collections: 0; major collections: 0
```

**Comment by miniHive** [comment]
I installed **datalog_cli** as recommended, via: "opam install datalog".

This is what I get:

% start datalog
**<snip>/datalog_cli: unknown option '--pattern'.**
compute fixpoint of given files
  -progress print progress
  -input print input clauses
  -output print facts after fixpoint
  -saturated print facts and clauses after fixpoint
  -sum aggregate number of literals for the given symbol
  **-pattern print facts matching this pattern**
  -goal add a goal for backward chaining
  -explain explain facts matching this pattern
  -query execute the query once fixpoint is reached
  -size print number of clauses after fixpoint
  -version print version
  -help  Display this list of options
  --help  Display this list of options

**Comment by c-cube** [comment]
what's the installed version? is it 0.6?

**Comment by miniHive** [comment]
I have version : 0.5.2.

**Comment by c-cube** [comment]
that's before the CLI arguments were changed, see `git log` :-)

**Comment by miniHive** [comment]
How does one get the latest version?
It does not seem to work with opam:

**$ opam upgrade
Everything as up-to-date as possible (run with --verbose to show unavailable upgrades).

$ opam install datalog
[NOTE] Package datalog is already installed (current version is 0.5.2).**

**Comment by c-cube** [comment]
have you run `opam update` first? does `0.6` show in `opam info datalog` at all?

**Comment by miniHive** [comment]
Somehow, opam is not aware of the latest version being 0.6:

$ opam update

=-=- Updating package repositories =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[default] synchronized from https://opam.ocaml.org/1.2.2

$ opam info datalog
             package: datalog
             version: 0.5.2
          repository: default
        upstream-url: https://github.com/c-cube/datalog/archive/0.5.2.tar.gz
       upstream-kind: http
   upstream-checksum: d3165be0b6946d860904b708cc44bcde
            homepage: https://github.com/c-cube/datalog
         bug-reports: https://github.com/c-cube/datalog/issues
            dev-repo: git://github.com/c-cube/datalog
              author: Simon Cruanes <simon.cruanes@inria.fr>
             license: BSD-2-Clause
                 doc: http://cedeela.fr/~simon/software/datalog/index.html
                tags: datalog, relational, query, prolog
             depends: ocamlfind & ocamlbuild & num
   installed-version: 0.5.2 [system]
  **available-versions: 0.1, 0.2, 0.3, 0.3.1, 0.4, 0.4.1, 0.5, 0.5.1, 0.5.2**
         description: An in-memory datalog implementation for OCaml.

It features two main algorithm:
- bottom-up focuses on big sets of rules with small relations, with frequent
  updates of the relations. Therefore, it tries to achieve good behavior in
  presence of incremental modifications of the relations.
- top-down resembles prolog (and allows nested subterms). It handles
  stratified negation and only explores the part of the search space that
  is relevant to a given query.

**Comment by c-cube** [comment]
That's because you're using opam 1.2 and not opam 2. Opam 1.2 has been deprecated for quite a while and it doesn't understand the packages of opam2.
